Libraries/PeriphDrivers/Source/UART/uart_reva.c

```diff
@@ -617,12 +685,62 @@ unsigned int MXC_UART_RevA_WriteTXFIFO(mxc_uart_reva_regs_t *uart, unsigned char
     return written;
 }
 
+int MXC_UART_RevA_SetAutoDMAHandlers(mxc_uart_reva_regs_t *uart, bool enable)
+{
+    int n = MXC_UART_GET_IDX((mxc_uart_regs_t *)uart);
+    MXC_ASSERT(n >= 0);
+
+    states[n].auto_dma_handlers = enable;
+
+    return E_NO_ERROR;
+}
+
+int MXC_UART_RevA_SetTXDMAChannel(mxc_uart_reva_regs_t *uart, unsigned int channel)
+{
+    int n = MXC_UART_GET_IDX((mxc_uart_regs_t *)uart);
+    if (n < 0)
```

Comment:
Here is another place where ASSERT may be better.  I see other places (not just in the code this PR touches) that could be ASSERTs also.  IMO, anything that checks to see if a valid UART instance was passed in is better as an ASSERT.

Reply:
https://github.com/Analog-Devices-MSDK/msdk/pull/763/commits/34db845346d0cd3b2dc574ea0b2dffcc4570d175

Reply:
@lorne-maxim I ended up purging the asserts except for once in the Init functions.  The compiler should catch those cases as type-cast mismatch, so I think the checks for a valid UART instance in every function were overkill anyways...

It reduces our kid size pretty significantly, which is nice